integreat_cms/cms/utils/slug_utils.py

```diff
@@ -169,6 +170,24 @@ def generate_unique_slug(**kwargs: Unpack[SlugKwargs]) -> str:
                 # the current object is also allowed to have the same slug
                 other_objects = other_objects.exclude(id=object_instance.id)
         if not other_objects.exists():
+            if (
+                foreign_model in ["page"]
+                and unique_slug in settings.RESERVED_REGION_PAGE_PATTERNS
+            ):
+                raise ValidationError(
+                    _("%(slug)s is a reserved slug."),
+                    code="invalid",
+                    params={"slug": unique_slug},
+                )
+            if (
+                foreign_model in ["region"]
+                and unique_slug in settings.RESERVED_REGION_SLUGS
+            ):
+                raise ValidationError(
+                    _("%(slug)s is a reserved slug."),
+                    code="invalid",
+                    params={"slug": unique_slug}
+                )
```

Comment:
```suggestion
        if not (
            other_objects.exists()
            or (
                foreign_model == "page"
                and unique_slug in settings.RESERVED_REGION_PAGE_PATTERNS
            )
            or (
                foreign_model == "region"
                and unique_slug in settings.RESERVED_REGION_SLUGS
            )
        ):
```
If you move the check into the slug helper, then I think it does not make sense to throw an error – just appending a number seems much more user friendly, as @david-venhoff suggested.
Then we also don't have to argue about formatting strings :sweat_smile: